extensions/wikia/MonetizationModule/MonetizationModuleHelper.class.php

```diff
@@ -152,22 +145,21 @@ public function getMonetizationUnits( $params ) {
 			];
 			$log->debug( "MonetizationModule: cannot get monetization units.", $loggingParams );
 		} else if ( !empty( $result ) ) {
-			$result = $this->processData( $result, $params );
+			$result = $this->processData( $title, $result, $params );
 		}
 
-		wfProfileOut( __METHOD__ );
-
 		return $result;
 	}
 
 	/**
 	 * Process data for the ad units (include logic to handle the data)
+	 * @param Title $title
 	 * @param string $data - data from API (json format)
 	 * @param array $params - API parameters
 	 * @param boolean $setMemc - set to true to set to memcache
 	 * @return mixed
 	 */
-	public function processData( $data, $params, $setMemc = true ) {
+	public function processData( Title $title, $data, $params, $setMemc = true ) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce the number of returns of this function 4, down to the maximum allowed 3. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1142)